internal/storage/dbrepository/DBRepository.go

```diff
@@ -0,0 +1,287 @@
+package dbrepository
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/dmitrovia/collector-metrics/internal/models/bizmodels"
+	"github.com/jackc/pgx/v5"
+)
+
+type DBepository struct {
+	databaseDSN   string
+	waitSecRespDB time.Duration
+	conn          *pgx.Conn
+}
+
+func (m *DBepository) Initiate(
+	dsn string,
+	waitSecRespDB time.Duration,
+	conn *pgx.Conn,
+) {
+	m.waitSecRespDB = waitSecRespDB
+	m.databaseDSN = dsn
+	m.conn = conn
+}
+
+func (m *DBepository) Init() {
+}
+
+func (m *DBepository) AddMetrics(
+	gauges map[string]bizmodels.Gauge,
+	counters map[string]bizmodels.Counter,
+) error {
+	ctx, cancel := context.WithTimeout(
+		context.Background(),
```

Comment:
[CHANGE REQUEST] It should be a parameter of this function.

Reply:
context transferred to function parameter